sklearn/utils/validation.py

```diff
@@ -407,19 +407,30 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
         dimensions or is originally 1D and ``ensure_2d`` is True. Setting to 0
         disables this check.
 
-    warn_on_dtype : boolean (default=False)
+    warn_on_dtype : boolean or None, optional (default=None)
         Raise DataConversionWarning if the dtype of the input data structure
         does not match the requested dtype, causing a memory copy.
 
+        .. deprecated:: 0.21
+            ``warn_on_dtype`` is deprecated in
+            version 0.21 and will be removed in 0.23.
```

Comment:
PEP8 says the limit is at 79 so there is no need to break that early:

```
        .. deprecated:: 0.21
            ``warn_on_dtype`` is deprecated in version 0.21 and will be
             removed in 0.23.
```

Reply:
done